torch/distributed/_tensor/api.py

```diff
@@ -242,6 +242,7 @@ def __new__(
             device=local_tensor.device,
             layout=local_tensor.layout,
             requires_grad=requires_grad,
+            data_ptr=local_tensor.data_ptr(),
```

Comment:
Since the current wrapper dataclass data_ptr always point to meaningless memory location. I think it seems better for us to point it to local tensor storage ptr, as it gives additional benefits to make apex kernels work :)  

Could you add some test in `test_dtensor.py` to assure that the code is working? i.e. after creating a DTensor, assert the `DTensor.data_ptr() == DTensor.to_local().data_ptr()`

Reply:
ok